ui/apps/platform/cypress/integration/risk/Risk.helpers.js

```diff
@@ -0,0 +1,169 @@
+    [deploymentscountAlias]: {
+        method: 'GET',
+        url: '/v1/deploymentscount*',
+    },
+    [searchOptionsAlias]: {
+        method: 'POST',
+        url: graphql('searchOptions'),
+    },
+};
+
+export function visitRiskDeployments() {
+    visit(riskURL, routeMatcherMap);
+
+    cy.get('h1:contains("Risk")');
+}
+
+export function visitRiskDeploymentsWithSearchQuery(search) {
+    visit(`${riskURL}${search}`, routeMatcherMap);
+
+    cy.get('h1:contains("Risk")');
+}
+
+export function viewRiskDeploymentByName(deploymentName) {
+    // Assume location is risk deployments table.
+    const routeMatcherMapForDeployment = {
+        'deploymentswithrisk/id': {
+            method: 'GET',
+            url: '/v1/deploymentswithrisk/*',
+        },
+    };
+
+    interactAndWaitForResponses(() => {
+        // Specify nth-child(4) for Namespace to make sure it is stackrox.
+        // For example, prevent multiple matches with collector in gmp-system namespace.
+        //
+        // Specify nth-child(1) for Name of deployment.
+        //
+        // Call contains method with RegExp for exact match and only first element.
+        // Unlike contains pseudo-selector which would require :nth(0) in case of multiple matches.
+        cy.get(
+            `.rt-tbody .rt-tr:has('.rt-td:nth-child(4):contains("stackrox")') .rt-td:nth-child(1)`
+        )
+            .contains(new RegExp(`^${deploymentName}$`))
+            .click();
+    }, routeMatcherMapForDeployment);
+
+    // Unlike some classic containers, risk list header has different data-testid attribute.
+    cy.get(`[data-testid="panel-header"]:contains("${deploymentName}")`);
+}
+
+export function viewRiskDeploymentInNetworkGraph() {
+    interactAndVisitNetworkGraphWithDeploymentSelected(() => {
+        cy.get('a:contains("View Deployment in Network Graph")').click();
+    });
+}
+
+// Process Discovery
+
+const getDeploymentEventTimelineAlias = 'getDeploymentEventTimeline';
+
+const routeMatcherMapForDeploymentEventTimeline = {
+    [getDeploymentEventTimelineAlias]: {
+        method: 'POST',
+        url: graphql('getDeploymentEventTimeline'),
+    },
+};
+
+export function viewGraph(fixtureForDeploymentEventTimeline) {
+    interactAndWaitForResponses(
+        () => {
+            cy.get('button:contains("View Graph")').click();
+        },
+        routeMatcherMapForDeploymentEventTimeline,
+        fixtureForDeploymentEventTimeline && {
+            [getDeploymentEventTimelineAlias]: {
+                fixture: fixtureForDeploymentEventTimeline,
+            },
+        }
+    );
+
+    cy.get('[data-testid="event-timeline"]');
+}
+
+const nextPageSelector = '[aria-label="Modal"] button[aria-label="Go to next page"]';
+
+export function clickNextPageInEventTimelineWithRequest(fixtureForDeploymentEventTimeline) {
+    interactAndWaitForResponses(
+        () => {
+            // Risk deployments list also has a button.
+            cy.get(nextPageSelector).click();
```

Comment:
What does this comment mean?

Reply:
Good eyes! I forgot to move the comment up when I factored out `nextPageSelector` above.

It was a too quickly written reminder why I added `[aria-label="Modal"]` because cypress found multiple matches with pagination button on the list page that is under the modal.

Reply:
I will move and improve in a follow up when I remove obsolete without mock API distinction in risk.test.js file.